Add more debug logs for cache leaks.

sipsma opened the pull request:
@tonistiigi trying to debug some errors a user is hitting where starting a container with a cache mount fails with errors like this: `runc run failed: unable to start container process: error during container init: error mounting "/var/lib/dagger/runc-overlayfs/snapshots/snapshots/347127/fs" to rootfs at "/var/lib/docker": stat /var/lib/dagger/runc-overlayfs/snapshots/snapshots/347127/fs: no such file or directory`

Haven't been able to repro on my own yet, so more detailed logs will be extremely helpful here. Probably for any other cache ref leaks in general too.

tonistiigi reviewed (commented):
How many log messages that is for a 10 step build approximately?

Wondering if some of them should go to "trace" level. That reminds me that we also need to move the healtcheck log to "trace" level.

sipsma commented:
@tonistiigi Good call on trace, here's what it looked like when running the `TestReproSourceDateEpoch` dockerfile test (first one I found with a non-trivial amount of steps): https://gist.github.com/sipsma/99b91b41127fa62b27e74728c667a5f9

So pretty noisy. I changed all of the noisy logs I added here to be Trace now, left one non-noisy one as debug. Also added corresponding `trace` flags and config file settings alongside the existing `debug` one.

I also figured that if we are willing to be very noisy on this trace level, may as well also include stack traces of the caller for these logs. They are huge and hard to read inline in the logs, but extremely helpful when debugging a potential cache leak.

I am not sure, but I think there could plausibly be performance overhead to calling `debug.Stack()` that often, so I also added `bklog.LazyStackTrace` which lets you include the stack trace as a field such that it will only actually be called if the log level is enabled (as opposed to always calling it but only showing it depending on the level).

---

Also went ahead and moved successful health checks to trace (left others w/ errors as is).

sipsma commented on solver/llbsolver/mounts/mount.go:
```diff
@@ -50,7 +50,7 @@ func (mm *MountManager) getRefCacheDir(ctx context.Context, ref cache.ImmutableR
 		cm:              mm.cm,
 		globalCacheRefs: sharedCacheRefs,
-		name:            fmt.Sprintf("cached mount %s from %s", m.Dest, mm.managerName),
+		name:            fmt.Sprintf("cached mount %s from %s with id %q", m.Dest, mm.managerName, id),
```
@tonistiigi before I press merge, just want to confirm this didn't get lost in the PR. It's technically not a new log, it's changing the name you see when you call disk usage to also include the actual id of the cache mount as opposed to just its mount location.

I figured there's no expectation of stability when it comes to these names, so this should be fine, just wanted to double check.

tonistiigi replied:
sgtm

tonistiigi replied:
Only thing is that I believe by default ID and `dest` are the same value. If that is the case would be nice to show ID only if it is different.

sipsma replied:
Oh sure, I forgot that's the dockerfile behavior, will fix before merge
